pkg/resource/managed_reconciler.go

```diff
@@ -341,6 +365,14 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 		return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	if err := r.managed.Initialize(ctx, managed); err != nil {
```

Comment:
Trying to work through the case where someone modifies the name of the external resource directly in the provider UI or something after a claim is already bound to a managed resource. If we only propagate the external name to the claim at "bind time" does that mean that the external name will not be propagated after the first time it is set?

Reply:
The definition of the external name is basically what provider uses as identifier. TBH, I can't see a scenario where the identifier can be changed unless you delete and re-create the resource.